tests/integ/sagemaker/workflow/test_workflow.py

```diff
@@ -1310,3 +1320,192 @@ def test_caching_behavior(
         except Exception:
             os.remove(script_dir + "/dummy_script.py")
             pass
+
+
+def test_multi_step_framework_processing_pipeline_uploads(
```

Comment:
* add a test with multiple framework processors but they don't share the same source_dir, or source_dir is not specified. 
* I think we can remove this: https://github.com/aws/sagemaker-python-sdk/blame/master/tests/integ/sagemaker/workflow/test_processing_steps.py#L284 because SKLearnProcessor is not a subclass of FrameworkProcessor

Reply:
Two tests have been added one with two FrameworkProcessor steps sharing the same local source_dir, the other with two FrameworkProcessor steps with different local source_dirs. I think we can add the improvements for other tests and use cases and in additional PRs as part of caching improvements part 2.